src/components/RestaurantItems/index.ts

```diff
@@ -3,26 +3,50 @@ import { Restaurant } from '../../types';
 import { $ } from '../../utils/dom';
 
 class RestaurantItems extends HTMLElement {
-  constructor() {
-    super();
-  }
-
   connectedCallback() {
     this.innerHTML = $template;
   }
 
   render(restaurants: Restaurant[]) {
     this.innerHTML = $template;
-    const $restaurantList = $<HTMLElement>('.restaurant-list');
+    const $restaurantList = $<HTMLElement>('.restaurant-list', this);
 
     restaurants.forEach((restaurant) => {
       $restaurantList.insertAdjacentHTML('beforeend', this.createRestaurantItem(restaurant));
     });
   }
 
   createRestaurantItem(restaurant: Restaurant) {
-    const { category, name, distance, description } = restaurant;
-    return `<restaurant-item category=${category} name=${name} distance=${distance} description=${description}></restaurant-item>`;
+    const { category, name, distance, description, link, favorite, id } = restaurant;
+    return `<restaurant-item category=${category} name=${name} distance=${distance} description=${description} favorite=${favorite} id=${id} link=${link} ></restaurant-item>`;
+  }
+
+  addBottomSheetHandler(handler: CallableFunction) {
+    this.addEventListener('click', (e: any) => {
+      if (e.target.className === 'favorite-icon') return;
+      const target = e.target.closest('restaurant-item');
+      const restaurant = [
+        'category',
+        'name',
+        'distance',
+        'description',
+        'link',
+        'favorite',
+        'id',
+      ].reduce((acc, cur) => {
+        acc[cur] = target.getAttribute(cur);
+        return acc;
+      }, {} as Record<string, string>);
```

Comment:
as 없이 한번 해보는거에 도전하면 어떨까요? 😄

Reply:
```suggestion
      ].reduce<Record<string, string>>((acc, cur) => {
        acc[cur] = target.getAttribute(cur);
        return acc;
      }, {});
```
이렇게 작성해볼 수 있을 것 같습니다!